File: Term2/Homework2/Kruskal/lib/Edge.cs
namespace Kruskal;

public class Edge : IComparable
{
	internal int From {get; init;}

	internal int To {get; init;}

	internal int Weight {get;}

	public Edge(int from, int to, int weight) 
	{
		this.From = from;
		this.To = to;
		this.Weight = weight;
	}

	public int CompareTo(object? obj) 
	{
		if (obj is not Edge || obj == null)
			return 0;
		
		Edge other = obj as Edge;
		if (this.Weight == other.Weight)
			return 0;
		else if (this.Weight > other.Weight)
			return 1;
		return -1;
	}
}

File: Term2/Homework2/Prim/src/main/java/utils/Edge.java
package utils;

public class Edge implements Comparable {
	protected int from;
	protected int to;
	protected int weight;

	public Edge(int from, int to, int weight) {
		this.from = from;
		this.to = to;
		this.weight = weight;
	}

	public int getFrom() {
		return from;
	}

	public int getTo() {
		return to;
	}

	public int getWeight() {
		return weight;
	}

	@Override
	public int compareTo(Object o) {
		if (o instanceof Edge || o == null) {
			Edge other = (Edge) o;
			return Integer.compare(this.weight, other.weight);
		}

		return 0;
	}

	@Override
	public boolean equals(Object o) {
		if (o instanceof Edge other) {
			return this.from == other.from && this.to == other.to && this.weight == other.weight;
		}

		return false;
	}
}